### DISCUSSION TITLE
How do we scale DragUI?

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [microsoft/microsoft-ui-xaml#10157] [Q&A]
If I'm using a SoftwareBitmap, how do I scale it to the correct size when used for a DragUI? I can call...

`args.DragUI.SetContentFromSoftwareBitmap(ImageMaster, offset);`

But it's the wrong size, it's rendering with the master images actual size. In practice, the image must be scaled to some size. I don't see any way to scale the image at render time.  There are suggestions to re-generate or recode the images, but that just seems insane to me when we can simply transform the texture. If we can't render to any size ... I have to keep a separate image for every size I want to render?

**Comment by castorix** [comment]
From my tests, I could not resize with async operations inside DragStarting
In this test, I used _BitmapEncoder_ (_BitmapTransform_) to resize the butterfly, but as it is async (with await), I could only call it outside DragStarting :
(or maybe it can be done with GDI/GDI+ to resize synchronously the SoftwareBitmap inside DragStarting, not tested)

![ScaledSoftwareBitmap](https://github.com/user-attachments/assets/284e3e4e-3d2c-4178-9214-d333457d3601)

**Reply by Gavin-Williams** [reply]
Ok. It doesn't seem natural to me. But I suppose I'm just going to regenerate all the textures every time the control changes size. I'll make a note that there might be a way to resize using gdi.
